test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests/Utility/AuditUtilityTests.cs

```diff
@@ -431,6 +453,7 @@ await createGraphTasks[1]
             "/path/proj.csproj",
             graphs,
             vulnerabilityProviders,
+            targetFrameworks: new List<TargetFrameworkInformation>(0),
```

Comment:
This test has a restore graph with two TFMs, so creating this collection with none means the production code is being tested with inputs that are impossible from a customer's perspective.  While I don't believe that this is causing any problems in this PR, the point is that an earlier iteration of this PR introduced a regression, even though all tests passed.

If we don't improve the test infrastructure, maybe the next person who works on `AuditUtility` will introduce a similar regression that tests won't catch, and whoever reviews that PR won't notice either.

Whereas, if we write our code in a way that prevents us from making these mistakes, then it's less likely that human mistakes will cause bugs in what we ship to customers.